lib/core/src/Cardano/Wallet/DB/Layer.hs

```diff
@@ -0,0 +1,813 @@
+            [ (TxMetaSlot >=.) <$> W.inclusiveLowerBound range
+            , (TxMetaSlot <=.) <$> W.inclusiveUpperBound range
+            , (TxMetaStatus ==.) <$> status
+            ]
+
+        , putLocalTxSubmission = \wid txid tx sl -> ExceptT $ do
+            let errNoSuchWallet = ErrPutLocalTxSubmissionNoSuchWallet $
+                    ErrNoSuchWallet wid
+            let errNoSuchTx = ErrPutLocalTxSubmissionNoSuchTransaction $
+                    ErrNoSuchTransaction wid txid
+
+            selectWallet wid >>= \case
+                Nothing -> pure $ Left errNoSuchWallet
+                Just _ -> handleConstraint errNoSuchTx $ do
+                    let record = LocalTxSubmission (TxId txid) wid sl tx
+                    void $ upsert record [ LocalTxSubmissionLastSlot =. sl ]
+
+        , readLocalTxSubmissionPending =
+            fmap (map localTxSubmissionFromEntity)
+            . listPendingLocalTxSubmissionQuery
+
+        , updatePendingTxForExpiry = \wid tip -> ExceptT $
+            selectWallet wid >>= \case
+        Nothing -> pure $ Left $ ErrNoSuchWallet wid
+        Just _ -> Right <$> updatePendingTxForExpiryQuery wid tip
+
+        , removePendingOrExpiredTx = \wid tid -> ExceptT $ do
+            let errNoSuchWallet =
+                    Left $ ErrRemoveTxNoSuchWallet $ ErrNoSuchWallet wid
+            let errNoMorePending =
+                    Left $ ErrRemoveTxAlreadyInLedger tid
+            let errNoSuchTransaction =
+                    Left $ ErrRemoveTxNoSuchTransaction $
+                    ErrNoSuchTransaction wid tid
+            selectWallet wid >>= \case
+                Nothing -> pure errNoSuchWallet
+                Just _  -> selectTxMeta wid tid >>= \case
+                    Nothing -> pure errNoSuchTransaction
+                    Just _ -> do
+                        count <- deletePendingOrExpiredTx wid tid
+                        pure $ if count == 0
+                            then errNoMorePending
+                            else Right ()
+
+        , getTx = \wid tid -> ExceptT $
+            readCheckpoint_ wid >>= \case
+        Nothing -> pure $ Left $ ErrNoSuchWallet wid
+        Just cp -> do
+            metas <- selectTxHistory cp
+                ti wid Nothing W.Descending
+                    [ TxMetaTxId ==. TxId tid ]
+            case metas of
+                [] -> pure (Right Nothing)
+                meta:_ -> pure (Right $ Just meta)
+
+        {-----------------------------------------------------------------------
+                                       Keystore
+        -----------------------------------------------------------------------}
+
+        , putPrivateKey = \wid key -> ExceptT $
+            selectWallet wid >>= \case
+        Nothing -> pure $ Left $ ErrNoSuchWallet wid
+        Just _ -> Right <$> do
+            deleteWhere [PrivateKeyWalletId ==. wid]
+            insert_ (mkPrivateKeyEntity wid key)
```

Comment:
Hi @paolino

Is there any reason for the reformatting of this block? It makes the case statement quite hard to understand, because the case matches are indented **_to the left_** of the case. But typically in the wallet, we indent case matches one block (4 spaces) **_to the right_** of a case.

The original code was formatted like this:
```hs
          , putPrivateKey = \wid key -> ExceptT $ do
              selectWallet wid >>= \case
                  Nothing -> pure $ Left $ ErrNoSuchWallet wid
                  Just _ -> Right <$> do
                      deleteWhere [PrivateKeyWalletId ==. wid]
                      insert_ (mkPrivateKeyEntity wid key)
```

Notice that the `Nothing` is indented **_to the right_** of the case.

Please would it be possible to go through this PR and remove any reformattings of already-correctly-formatted code?

Many thanks!